hw/drivers/sensors/lis2ds12/src/lis2ds12.c

```diff
@@ -0,0 +1,2723 @@
+ *
+ * @param the sensor interface
+ * @param pointer to return wake_up_src in
+ * @return 0 on success, non-zero on failure
+ */
+int lis2ds12_get_wake_up_src(struct sensor_itf *itf, uint8_t *status)
+{
+    return lis2ds12_read8(itf, LIS2DS12_REG_WAKE_UP_SRC, status);
+}
+
+/**
+ * Get Tap Source
+ *
+ * @param the sensor interface
+ * @param pointer to return tap_src in
+ * @return 0 on success, non-zero on failure
+ */
+int lis2ds12_get_tap_src(struct sensor_itf *itf, uint8_t *status)
+{
+    return lis2ds12_read8(itf, LIS2DS12_REG_TAP_SRC, status);
+}
+
+/**
+ * Get 6D Source
+ *
+ * @param the sensor interface
+ * @param pointer to return sixd_src in
+ * @return 0 on success, non-zero on failure
+ */
+int lis2ds12_get_sixd_src(struct sensor_itf *itf, uint8_t *status)
+{
+    return lis2ds12_read8(itf, LIS2DS12_REG_6D_SRC, status);
+}
+
+/**
+ * Set whether interrupt 2 signals is mapped onto interrupt 1 pin
+ *
+ * @param the sensor interface
+ * @param value to set (false = disabled, true = enabled)
+ * @return 0 on success, non-zero on failure
+ */
+int lis2ds12_set_int2_on_int1_map(struct sensor_itf *itf, bool enable)
+{
+    uint8_t reg;
+    int rc;
+
+    rc = lis2ds12_read8(itf, LIS2DS12_REG_CTRL_REG5, &reg);
+    if (rc) {
+        return rc;
+    }
+
+    if (enable) {
+        reg |= LIS2DS12_CTRL_REG5_INT2_ON_INT1;
+    } else {
+        reg &= ~LIS2DS12_CTRL_REG5_INT2_ON_INT1;
+    }
+
+    return lis2ds12_write8(itf, LIS2DS12_REG_CTRL_REG5, reg);
+}
+
+/**
+ * Get whether interrupt 2 signals is mapped onto interrupt 1 pin
+ *
+ * @param the sensor interface
+ * @param value to set (0 = disabled, 1 = enabled)
+ * @return 0 on success, non-zero on failure
+ */
+int lis2ds12_get_int2_on_int1_map(struct sensor_itf *itf, uint8_t *val)
+{
+    uint8_t reg;
+    int rc;
+
+    rc = lis2ds12_read8(itf, LIS2DS12_REG_CTRL_REG5, &reg);
+    if (rc) {
+        return rc;
+    }
+
+    *val = (reg & LIS2DS12_CTRL_REG5_INT2_ON_INT1) >> 5;
+    return 0;
+}
+
+/**
+ * Run Self test on sensor
+ *
+ * @param the sensor interface
+ * @param pointer to return test result in (0 on pass, non-zero on failure)
+ *
+ * @return 0 on sucess, non-zero on failure
+ */
+int lis2ds12_run_self_test(struct sensor_itf *itf, int *result)
+{
+    int rc;
+
+    int16_t no_st[3], st[3], data[3] = {0,0,0};
+    int32_t scratch[3] = {0,0,0};
+    int i;
+    uint8_t prev_config[6];
+    *result = 0;
+    uint8_t config[6] = { LIS2DS12_DATA_RATE_HR_14BIT_50HZ | LIS2DS12_FS_2G | LIS2DS12_CTRL_REG1_BDU, LIS2DS12_CTRL_REG2_IF_ADD_INC, 0, 0, 0, 0};
+    uint8_t fs;
+
+    rc = lis2ds12_readlen(itf, LIS2DS12_REG_CTRL_REG1, prev_config, 6);
+    if (rc) {
+        return rc;
+    }
+
+    rc = lis2ds12_writelen(itf, LIS2DS12_REG_CTRL_REG2, &config[1], 5);
+    if (rc) {
+        return rc;
+    }
+
+    rc = lis2ds12_write8(itf, LIS2DS12_REG_CTRL_REG1, config[0]);
+    if (rc) {
+        return rc;
+    }
+
+    /* wait 200ms */
+    os_time_delay(OS_TICKS_PER_SEC / 200);
```

Comment:
OS_TICKS_PER_SEC / 5 is 200msec. Assuming the comment is right then code is wrong :) There's few other spots within this routine which have the same thing.